src/Nethermind/Nethermind.Consensus/Producers/BlockProducerBase.cs

```diff
@@ -186,23 +186,6 @@ protected BlockProducerBase(
             return Task.FromResult((Block?)null);
         }
 
-        /// <summary>
-        /// Sets the state to produce block on
-        /// </summary>
-        /// <param name="parentStateRoot">Parent block state</param>
-        /// <returns>True if succeeded, false otherwise</returns>
-        /// <remarks>Should be called inside <see cref="_producingBlockLock"/> lock.</remarks>
-        protected bool TrySetState(BlockHeader? parent)
-        {
-            if (parent is not null && StateProvider.HasStateForBlock(parent))
-            {
-                StateProvider.SetBaseBlock(parent);
```

Comment:
Uses `BlochainProcessor` which call `BlockProcessor` which calls `SetBaseBlock` so this is unnecessary.